common/src/main/java/com/wynntils/overlays/SpellInputsOverlay.java

```diff
@@ -0,0 +1,193 @@
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import java.util.Arrays;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.DeltaTracker;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.Style;
+import net.minecraft.resources.ResourceLocation;
+import net.neoforged.bus.api.SubscribeEvent;
+
+public class SpellInputsOverlay extends Overlay {
+    // Upon reaching level 11, the spell inputs are displayed using the smaller icon
+    private static final int SMALL_CHARACTERS_LEVEL = 11;
+    private static final ResourceLocation SPELL_INPUTS_FONT =
+            ResourceLocation.withDefaultNamespace("hud/gameplay/default/bottom_middle");
+
+    private static final String FULL_NO_CLICK_CHARACTER = "\uE102";
+    private static final String SMALL_NO_CLICK_CHARACTER = "\uE105";
+    private static final String SEPARATOR_CHARACTER = "\uE106";
+
+    @Persisted
+    private final Config<Boolean> shouldDisplayOriginal = new Config<>(false);
+
+    @Persisted
+    private final Config<SpellInputStyle> inputStyle = new Config<>(SpellInputStyle.ORIGINAL);
+
+    @Persisted
+    private final Config<TextShadow> textShadow = new Config<>(TextShadow.NONE);
+
+    @Persisted
+    protected final Config<Float> fontScale = new Config<>(1.0f);
+
+    private StyledText spellText = StyledText.EMPTY;
+
+    public SpellInputsOverlay() {
+        super(
+                new OverlayPosition(
+                        -62,
+                        0,
+                        VerticalAlignment.BOTTOM,
+                        HorizontalAlignment.CENTER,
+                        OverlayPosition.AnchorSection.BOTTOM_MIDDLE),
+                new OverlaySize(100, 20),
+                HorizontalAlignment.CENTER,
+                VerticalAlignment.BOTTOM);
+    }
+
+    @SubscribeEvent
+    public void onSpellCast(SpellEvent.Partial event) {
+        SpellDirection[] dirs = event.getSpellDirectionArray();
+        SpellInputStyle style = inputStyle.get();
+
+        spellText = switch (style) {
+            case SMALL -> buildUnicodeInputs(dirs, true);
+            case FULL -> buildUnicodeInputs(dirs, false);
+            case ORIGINAL -> buildUnicodeInputs(dirs, Models.CharacterStats.getLevel() >= SMALL_CHARACTERS_LEVEL);
+            case LEGACY -> buildLegacyInputs(dirs);
+        };
+    }
+
+    @SubscribeEvent
+    public void onSpellExpired(SpellEvent.Expired event) {
+        spellText = StyledText.EMPTY;
+    }
+
+    @Override
+    protected void onConfigUpdate(Config<?> config) {
+        Models.Spell.setHideSpellInputs(Managers.Overlay.isEnabled(this) && !this.shouldDisplayOriginal.get());
```

Comment:
Is this called at startup as well? Otherwise I don't see how the SpellModel could know that it should hide the original.